docs/_source/tutorials-examples/population-synthesis/custom_step_and_flow.py

```diff
@@ -0,0 +1,38 @@
+from posydon.binary_evol.flow_chart import (flow_chart, STAR_STATES_NORMALSTAR)
+
+def end_flow_chart(FLOW_CHART=None):
+    FLOW_CHART = {}
+    for s1 in STAR_STATES_NORMALSTAR:
+        for s2 in STAR_STATES_NORMALSTAR:
+            FLOW_CHART[(s1, s2, 'detached', 'ZAMS')] = 'step_end'
+    return FLOW_CHART
+
+
+
+class my_CE_step(object):
+    """Compute a fake CE event."""
+
+    def __init__(self, verbose=False):
+        self.verbose = verbose
+
+    def __call__(self, binary):
+
+        if self.verbose:
+            print('The orbital separation post CE is half the pre CE orbital separation!')
+
+        # Determine which star is the donor and which is the companion
+        if binary.event in ["oCE1", "oDoubleCE1"]:
+            donor_star = binary.star_1
+            comp_star = binary.star_2
+        elif binary.event in ["oCE2", "oDoubleCE2"]:
+            donor_star = binary.star_2
+            comp_star = binary.star_1
+        else:
+            raise ValueError("CEE does not apply if `event` is not "
+                            "`oCE1`, 'oDoubleCE1' or `oCE2`, 'oDoubleCE1'")
+
+        binary.orbital_period /= 2.
+        donor_star.mass = donor_star.he_core_mass # lose envelope
```

Comment:
Here is a check missing, that we have a `H-rich` star as donor, which has a well defined `he_core_mass`.

Reply:
I don't think we need a fully working step. It will show them failure too.

Reply:
OK, if this is an intended failure, we should document it this way. Otherwise people will think: their tutorial is failing.

Reply:
Added two errors.